src/Formik.tsx

```diff
@@ -384,20 +384,28 @@ export class Formik<Values = FormikValues> extends React.Component<
     value: any,
     shouldValidate: boolean = true
   ) => {
-    if (this.didMount) {
-      // Set form field by name
-      this.setState(
-        prevState => ({
-          ...prevState,
-          values: setIn(prevState.values, field, value),
-        }),
-        () => {
-          if (this.props.validateOnChange && shouldValidate) {
-            this.runValidations(this.state.values);
+    return new Promise((resolve, reject) => {
+      if (this.didMount) {
+        // Set form field by name
+        this.setState(
+          prevState => ({
+            ...prevState,
+            values: setIn(prevState.values, field, value),
+          }),
+          () => {
+            if (this.props.validateOnChange && shouldValidate) {
+              this.runValidations(this.state.values)
+                .then(resolve)
```

Comment:
```suggestion
                .then(() => resolve())
```
To make sure the promise returned resolves with the same result in all cases.
This introduces inconsistency in resolved values as the `runValidations` will resolve to `FormikErrors<Values>` which will be passed as a result of our promise while in other cases below our promise is resolved with no value (just `resolve()`).